src/plugins/intel_cpu/src/emitters/snippets/aarch64/utils.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <set>
+#include <vector>
+
+#include "cpu/aarch64/jit_generator.hpp"
+#include "snippets/emitter.hpp"
+#include "snippets/lowered/expression_port.hpp"
+
+namespace ov::intel_cpu::aarch64::utils {
+
+inline static std::vector<Xbyak_aarch64::XReg> transform_idxs_to_regs(const std::vector<size_t>& idxs) {
+    std::vector<Xbyak_aarch64::XReg> regs;
+    regs.reserve(idxs.size());
+    std::transform(idxs.begin(), idxs.end(), std::back_inserter(regs), [](size_t idx) {
+        return Xbyak_aarch64::XReg(static_cast<int>(idx));
+    });
+    return regs;
+}
+
+/**
+ * @brief Find the available register from the pool excepting: abi_param1, abi_param2, SP and `used_gpr_idxs`
+ * @param used_gpr_idxs current used gpr register indexes
+ * @return register
+ */
+Xbyak_aarch64::XReg get_aux_gpr(const std::vector<size_t>& used_gpr_idxs);
+
+/**
+ * @brief Returns aux gpr register for dynamic memory access emitters. Returns a register from `aux_gpr_idxs`.
```

Comment:
I think we don't need to specify that this is exactly for memory access emitters: this is a common helper which allows to get aux gpr (the mechanism is described in the description). Could you please reformulate this part a bit?

Reply:
Reformulated